renderdoc/driver/gl/wrappers/gl_buffer_funcs.cpp

```diff
@@ -2960,6 +2960,25 @@ bool WrappedOpenGL::Serialise_glVertexArrayVertexAttribOffsetEXT(GLuint vaobj, G
   return true;
 }
 
+std::function<void()> WrappedOpenGL::ConvertClientMemoryPointer(GLuint index, const void *&pointer)
+{
+  GLuint buffer;
+  m_Real.glGetIntegerv(eGL_VERTEX_ARRAY_BINDING, (GLint *)&buffer);
+
+  if(m_State >= WRITING && !buffer)
+  {
+    // Copy all client memory, and the pointer becomes a zero offset.
+    m_ClientMemoryPointers[index] = pointer;
+    pointer = nullptr;
+
+    // Bind the temporary VBO, and restore the 0-binding afterwards.
+    glBindBuffer(eGL_ARRAY_BUFFER, m_ClientMemoryVBOs[index]);
+    return [this]() { glBindBuffer(eGL_ARRAY_BUFFER, 0); };
```

Comment:
I know I dropped VS2010 very recently so lambdas will compile now, but this seems vastly overengineered. You can return a `bool` that indicates whether a temporary VBO was bound, and then do the `glBindBuffer` in the calling function.